crates/core/src/contracts/contract_account.rs

```diff
@@ -58,4 +58,23 @@ impl<'a, P: Provider + Send + Sync> ContractAccount<'a, P> {
         let value = Into::<U256>::into(low) + (Into::<U256>::into(high) << 128);
         Ok(value)
     }
+
+    /// Returns the nonce of the contract account.
+    pub async fn nonce(&self, block_id: &BlockId) -> Result<U256, EthApiError<P::Error>> {
+        // Prepare the calldata for the get_nonce function call
+        let calldata = vec![];
+        let request = FunctionCall { contract_address: self.address, entry_point_selector: GET_NONCE, calldata };
+
+        let result = self.provider.call(request, block_id).await?;
+        if result.len() != 1 {
+            return Err(DataDecodingError::InvalidReturnArrayLength {
+                entrypoint: "get_nonce".into(),
+                expected: 1,
+                actual: result.len(),
+            }
+            .into());
+        }
+
+        Ok(U256::from_be_bytes(result[0].to_bytes_be()))
```

Comment:
Cleaner to go through the felt252wrapper imo here, since it's our go-to wrapper for type conversion